source/common/router/config_impl.h

```diff
@@ -178,17 +179,19 @@ class VirtualHostImpl : public VirtualHost {
   HeaderParserPtr response_headers_parser_;
   PerFilterConfigs per_filter_configs_;
   const bool include_attempt_count_;
+  const envoy::api::v2::route::VirtualHost config_;
```

Comment:
Maybe store an `const absl::optional<RoutePolicy>` instead of storing the entire vhost config?

Reply:
I thought about going down that route but it had a bit more friction...

Reply:
These objects might end up quite large, so I think this optimization might make sense.